mergify_engine/date.py

```diff
@@ -261,3 +263,40 @@ def fromtimestamp(timestamp: float) -> datetime.datetime:
 
 def pretty_datetime(dt: datetime.datetime) -> str:
     return dt.strftime("%Y-%m-%d %H:%M %Z")
+
+
+_INTERVAL_RE = re.compile(
+    r"""
+    (?P<filled>
+        ((?P<days>[-+]?\d+) \s+ d(ays?)? \s* )?
+        ((?P<hours>[-+]?\d+) \s+ h(ours?)? \s* )?
+        ((?P<minutes>[-+]?\d+) \s+ m(inutes?)? \s* )?
+        ((?P<seconds>[-+]?\d+) \s+ s(econds?)? \s* )?
+    )
+    """,
+    re.VERBOSE,
+)
+
+
+def interval_from_string(
+    value: str, min_: typing.Optional[datetime.timedelta] = None
+) -> datetime.timedelta:
+    m = _INTERVAL_RE.match(value)
+    if m is None:
+        raise InvalidDate("Invalid date interval")
+
+    kw = typing.cast(TimedeltaRegexResultT, m.groupdict())
+    print(f"{value}: {kw} {m.groups()}")
+    if not kw or not kw["filled"]:
+        raise InvalidDate("Invalid date interval")
+
+    td = datetime.timedelta(
+        days=int(kw["days"] or 0),
+        hours=int(kw["hours"] or 0),
+        minutes=int(kw["minutes"] or 0),
+        seconds=int(kw["seconds"] or 0),
+    )
+    if min_ is not None and td < min_:
```

Comment:
Not a fan of having this kind of feature in a parser.
Not a parsing problem.

Reply:
I will move it in voluptuous side